ER-model

Johney9 opened the pull request.

stanislavgrujic commented on .idea/vcs.xml:
```diff
@@ -0,0 +1,6 @@
+  <component name="VcsDirectoryMappings">
+    <mapping directory="" vcs="Git" />
+  </component>
+</project>
```
This should go into the .gitignore

stanislavgrujic commented on uml/er-model/ERModel.puml:
```diff
@@ -0,0 +1,72 @@
+@startuml
```
I expected that you will create a full Java  Maven project which uses the `plantuml-maven-plugin` to generate the ER schema.
In that case, you would have hierarchy `src/main/resources/uml` to place these files.

stanislavgrujic commented:
Could you please attach generated screenshot, it will be easier to do the review?

Johney9 commented:
Changed project to full Maven, using plantuml-maven-plugin to generate the files. Added .idea folder to .ignore. Also added the mysql model and dependencies to pom.xml

![ermodel](https://user-images.githubusercontent.com/11011741/29657310-78fc3ab8-88b7-11e7-972c-7967408897a1.png)

stanislavgrujic commented:
What does file `src/main/resources/sql/model/team_pomodoro_db_model.mwb` represent? If it is generated, it should not be commit in the source code control.

Johney9 commented:
It represents the mysql model from which the db schema is generated. Should I remove it?

stanislavgrujic commented:
Which tool did you use to do it? Please remove the .bak, and this one depends on your answer. :)

Johney9 commented:
I used mysql workbench. Will remove .bak

stanislavgrujic commented:
PNG files should not be commit and should not go into the `src` folder, but should go into `target` folder, with `.class` files

stanislavgrujic commented on .gitignore:
```diff
@@ -20,3 +20,11 @@
 hs_err_pid*
+
+#IntelliJ IDE project folder
+src/test/
```
Why would you ignore the test files?

stanislavgrujic commented on .gitignore:
```diff
@@ -20,3 +20,11 @@
+
+#IntelliJ IDE project folder
+src/test/
+/src/main/resources/sql/scripts/schema_create_script.sql
```
Why is this file ignored, is it generated? If so, it should not go into `/src/main/resources` folder.

stanislavgrujic commented on pom.xml:
```diff
@@ -0,0 +1,109 @@
+                    <dependency>
+                        <groupId>net.sourceforge.plantuml</groupId>
+                        <artifactId>plantuml</artifactId>
+                        <version>8059</version>
```
There is a dependency version mismatch between this one and the one used in the project dependencies definition. Please extract `plantuml.version` and use the same version in all places.
